doc/nrf/ug_bootloader.rst

```diff
@@ -7,212 +7,185 @@ Secure bootloader chain
-In addition, it can provide functionality for upgrading both itself and the following image in the boot sequence (in most cases, the application).
-
-There are two partitions where the upgradable bootloader can be stored: slot 0 and slot 1 (also called *S0* and *S1*).
-A new bootloader image is placed in the partition that is currently not in use.
-When booting, the immutable bootloader checks the version information for the images in slot 0 and slot 1 and boots the one with the highest version.
-If this image is faulty and cannot be booted, the other partition will always hold a working image and be booted instead.
-
-Set the option :option:`CONFIG_BUILD_S1_VARIANT` when building the upgradable bootloader to automatically generate pre-signed variants of the image for both slots.
-These signed variants can be used to perform an upgrade procedure through the :ref:`lib_fota_download` library.
-
-.. _ug_bootloader_adding:
-
-Adding a bootloader chain to your application
-*********************************************
+Second-stage Upgradable bootloader
+==================================
 
-The |NCS| includes a sample implementation of an :ref:`immutable bootloader <bootloader>`.
-Additionally, the |NCS| comes with a slightly modified version of :doc:`mcuboot:index`.
+If you also need the capability of updating the bootloader, you can use a second-stage, upgradable bootloader.
+It can be updated through :ref:`firmware-over-the-air (FOTA) <lib_fota_download>` updates, unlike the immutable bootloader.
 
-Both bootloaders can easily be included in your application using :ref:`multi-image builds <ug_multi_image>`.
+The root of trust, in the form of the immutable bootloader, protects the upgradable bootloader, which must also continue the chain of trust by verifying the next image in the boot sequence.
+Because of this, the immutable bootloader is responsible for upgrading the upgradable bootloader and verifying its metadata and image integrity.
+For more information about how the immutable bootloader accomplishes this, see the :ref:`bootloader_flash_layout` section of the |NSIB|.
 
-Adding the immutable bootloader
-===============================
+The upgradable bootloader carries out the same signature and metadata verification as the immutable bootloader.
+Also, it can upgrade both itself and the following image in the boot sequence, which, in most cases, is an application.
 
-To add the immutable bootloader to your application, set :option:`CONFIG_SECURE_BOOT` and add your private key file under :option:`CONFIG_SB_SIGNING_KEY_FILE`.
-|how_to_configure|
+.. caution::
+   You should add a second-stage bootloader only when necessary by the design or firmware upgrade needs.
+   Adding the second stage bootloader for no reason will lead to a degradation of the system's overall security, as attackers can exploit bugs that may exist in either bootloader.
```

Comment:
very probably overall security level estimation of `NISB + sdk-mcuboot` is higher than standalone sdk-mcuboot as 2-stages solution gives capability of upgrade sdk-mcuboot (which has much bigger codebase than NISB and therefore might be more affected by bugs).

Reply:
I see your point here, but I think what @pdunaj was meaning by this caution is that more code in general means higher chance of bugs, even if one of those codebases is upgradable.